sky/skylet/providers/runpod/node_provider.py

```diff
@@ -0,0 +1,150 @@
+        """
+        nodes = self._get_filtered_nodes(tag_filters=tag_filters)
+        return [node_id for node_id, _ in nodes.items()]
+
+    def is_running(self, node_id):
+        """Return whether the specified node is running."""
+        return self._get_cached_node(node_id=node_id) is not None
+
+    def is_terminated(self, node_id):
+        """Return whether the specified node is terminated."""
+        return self._get_cached_node(node_id=node_id) is None
+
+    def node_tags(self, node_id):
+        """Returns the tags of the given node (string dict)."""
+        return self._get_cached_node(node_id=node_id)['tags']
+
+    def external_ip(self, node_id):
+        """Returns the external ip of the given node."""
+        return self._get_cached_node(node_id=node_id)['ip']
+
+    def internal_ip(self, node_id):
+        """Returns the internal ip (Ray ip) of the given node."""
+        return self._get_cached_node(node_id=node_id)['ip']
+
+    def create_node(self, node_config: Dict[str, Any], tags: Dict[str, str], count: int) -> Optional[Dict[str, Any]]:
+        """Creates a number of nodes within the namespace."""
+        # Get the tags
+        config_tags = node_config.get('tags', {}).copy()
+        config_tags.update(tags)
+        config_tags[TAG_RAY_CLUSTER_NAME] = self.cluster_name
+
+        # Create nodes
+        ttype = node_config['InstanceType']
+        region = self.provider_config['region']
+
+        for _ in range(count):
+            instance_id = runpod_api.launch(name=self.cluster_name,
+                                            instance_type=ttype,
+                                            region=region
+                                            )
+
+        if instance_id is None:
+            raise RunPodError('Failed to launch instance.')
+
+        runpod_api.set_tags(instance_id, config_tags)
+
+        instance_status = "PENDING"
+        while instance_status != "RUNNING":
+            instance_status = runpod_api.list_instances()[instance_id]['status']
+
+    @synchronized
+    def set_node_tags(self, node_id: str, tags: Dict[str, str]) -> None:
+        """Sets the tag values (string dict) for the specified node."""
+        node = self._get_node(node_id)
+        node['tags'].update(tags)
+        runpod_api.set_tags(node_id, node['tags'])
+
+    def terminate_node(self, node_id: str) -> Optional[Dict[str, Any]]:
+        """Terminates the specified node."""
+        runpod_api.remove(node_id)
+
+    @synchronized
+    def _get_filtered_nodes(self, tag_filters: Dict[str, str]) -> Dict[str, Any]:
+        '''SkyPilot Method
+        Caches the nodes with the given tag_filters.
+        '''
+        instances = runpod_api.list_instances()
+
+        new_cache = {}
+        for instance_id, instance in instances.items():
+            if instance['status'] != 'running':
+                continue
+            if any(tag in instance['tags'] for tag in tag_filters):
```

Comment:
What is the purpose of this check? It might worth a comment.

We should instead filter the nodes by the `self.cluster_name`:
```
if instance['name'] == self.cluster_name:
```